homeassistant/components/reolink/__init__.py

```diff
@@ -31,14 +34,15 @@ class ReolinkData:
     device_coordinator: DataUpdateCoordinator
 
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
     """Set up Reolink from a config entry."""
-    host = ReolinkHost(hass, dict(entry.data), dict(entry.options))
+    host = ReolinkHost(hass, dict(config_entry.data), dict(config_entry.options))
```

Comment:
If we don't need to modify the data or options we can pass them directly and change the type annotation of them to `Mapping` in `ReolinkHost`.

Reply:
The ReolinkHost is also initialized in the config_flow inside async_obtain_host_settings.
There dicts are used instead of Mapping, so one always needs to be converted.

Reply:
I meant that we can change the type annotation of the `ReolinkHost` parameters to `Mapping`. That will allow us to pass both `dict` and `MappingProxyType` instances.